en/components/transaction-how-to.md

```diff
@@ -0,0 +1,62 @@
+---
+title: igxBaseTransaction, igxTransaction and igxHierarchical transaction explained
+_description: 
+_keywords: Ignite UI for Angular, transaction
+---
+
+## igxBaseTransaction, igxTransaction and igxHierarchical transaction explained
+
+### Transaction, State, Transaction Log
+
+The Transaction is the main building block of the Transaction service. The Transaction is actually every operation that you execute on the data. The `Transaction` interface defines three properties: `id`, `newValue` and `type`.
+
+The `id` of the Transaction should be unique and defines the record that this transaction is affecting. The `type` may be any of the three transaction types: `ADD`, `DELETE` and `UPDATE`, depending what operation you execute. The `newValue` contains the value of the new record in case you are adding an `ADD` transaction. If you are editing an existing record, the `newValue` would contain the changes only. If you are deleting a record, the `newValue` will contain the value of the deleted record. 
+
+You may see an example of how adding each type of transaction looks like in the [How to use the Transaction service](transaction-example.md) topic.
+
+Every time you add a Transaction, it is added to the transaction log and undo stack. All the changes in the transaction log are then accumulated per record. From that point, the service maintains an aggregated `State`. The `State`consists of unique records and every record may be of one of the supported transaction types, mentioned above.
+
```

Comment:
We should include something about pending transactions here. Please add this or something better:

While adding transactions you may turn on pending transactions by calling `startPending`. All subsequent transactions will be accumulated in single transaction until you call `endPending`. If you pass `true` to `endPending` all accumulated  transactions will be added as a single transaction in transaction log and in undo stack.